cmd/login_cloud.go

```diff
@@ -14,37 +16,96 @@ var loginCloudCommand = &cobra.Command{
 	Short: "Authenticate with Load Impact",
 	Long: `Authenticate with Load Impact.
 
-This will set the default server used when just "-o cloud" is passed.`,
+This will set the default token used when just "k6 run -o cloud" is passed.`,
+
+	Example: `
+  # Show the stored token.
+  k6 login cloud -s
+
+  # Set up the token.
+  k6 login cloud -t YOUR_TOKEN
+
+  # Ask for your Load Impact user email and password to automatically set up the token.
+  k6 login cloud`[1:],
+
 	Args: cobra.NoArgs,
 	RunE: func(cmd *cobra.Command, args []string) error {
+
+		printToken := func(conf cloud.Config) {
+			fmt.Fprintf(stdout, "  token: %s\n", ui.ValueColor.Sprint(conf.Token))
+		}
+
 		config, cdir, err := readDiskConfig()
 		if err != nil {
 			return err
 		}
 
-		conf := config.Collectors.Cloud
-		form := ui.Form{
-			Fields: []ui.Field{
-				ui.StringField{
-					Key:     "token",
-					Label:   "API Token",
-					Default: conf.Token,
-				},
-			},
-		}
-		vals, err := form.Run(os.Stdin, stdout)
+		flags := cmd.Flags()
+		show, err := flags.GetBool("show")
 		if err != nil {
 			return err
 		}
-		if err := mapstructure.Decode(vals, &conf); err != nil {
+		token, err := flags.GetString("token")
+		if err != nil {
 			return err
 		}
 
+		conf := config.Collectors.Cloud
+
+		switch {
+		case show:
+			printToken(conf)
+			return nil
+		case token != "":
+			conf.Token = token
+		default:
+			printToken(conf)
+
+			form := ui.Form{
+				Fields: []ui.Field{
+					ui.StringField{
+						Key:   "Email",
+						Label: "Email",
+					},
+					ui.StringField{
+						Key:   "Password",
+						Label: "Password",
+					},
+				},
+			}
+
+			vals, err := form.Run(os.Stdin, stdout)
+			if err != nil {
+				return err
+			}
+
+			email := vals["Email"].(string)
```

Comment:
Nitpick: I'd group `email` and `password` with the `form.Run()` above instead of with the response stuff.